lib/datapacksjob.js

```diff
@@ -2591,3 +2598,130 @@ DataPacksJob.prototype.retrieveSalesforce = async function(jobInfo) {
 DataPacksJob.prototype.deploySalesforce = async function(jobInfo) {
     await this.vlocity.datapacksutils.deploySalesforce(jobInfo);
 }
+
+DataPacksJob.prototype.rmdirCustom = async function(dir) {
+	var list = fs.readdirSync(dir);
+	for(var i = 0; i < list.length; i++) {
+		var filename = path.join(dir, list[i]);
+		var stat = fs.statSync(filename);
+
+		if(filename == "." || filename == "..") {
+			// pass these files
+		} else if(stat.isDirectory()) {
+			// rmdir recursively
+			this.rmdirCustom(filename);
+		} else {
+			// rm fiilename
+			fs.unlinkSync(filename);
+		}
+	}
+	fs.rmdirSync(dir);
+};
+
+DataPacksJob.prototype.downloadVplDatapack = async function (jobInfo) {
+    var deletedOnce =  false;
+    let VplAssetRecords;
+    let vplAttachmentRecords;
+    let vplName = jobInfo.VplName;
+    try {
+
+        if (vplName.includes("---")) {
+            vplName = vplName.split("---").join("/");
+        }
+
+        let query = "SELECT VPL_Asset__c FROM VlocityProcessAssetAssociation__c WHERE VPL_Listing__r.Name = '"+ vplName +"' AND VPL_Asset__r.Type__c='JSON DataPack' AND VPL_Asset__r.Active__c=true";
+        VlocityUtils.verbose('query:',query);
+        await this.vlocity.jsForceConnection.query(query, function (err, result) {
+            if (err) {
+                jobInfo.hasError = true;
+                jobInfo.errors.push(err);
+            }
+            VplAssetRecords = result.records;
+            VlocityUtils.verbose("total-2 : " + result.totalSize);
+            VlocityUtils.verbose("fetched-2 : " + result.records.length);
+            VlocityUtils.verbose(VplAssetRecords);
+            jobInfo.data = result;
+        });
+
+        for(res in VplAssetRecords)  {
+            let query = "SELECT Name, Body FROM Attachment WHERE ParentId = '" + VplAssetRecords[res].VPL_Asset__c + "'";
+            VlocityUtils.verbose('query:');
+            VlocityUtils.verbose(query);
+            await this.vlocity.jsForceConnection.query(query, function (err, result) {
```

Comment:
You should use the try / catch syntax for these instead of a callback. Better for readability.

Reply:
ok